osgp/platform/osgp-adapter-kafka-distributionautomation/src/test/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/LowVoltageMessageFactory.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.kafka.da.application.mapping;
+
+import com.alliander.data.scadameasurementpublishedevent.Analog;
+import com.alliander.data.scadameasurementpublishedevent.UnitMultiplier;
+import com.alliander.data.scadameasurementpublishedevent.UnitSymbol;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LowVoltageMessageFactory extends MessageFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(LowVoltageMessageFactory.class);
+
+  public enum Version {
+    VERSION_1,
+    VERSION_2;
+  }
+
+  public static List<Analog> expectedMeasurements(final Version version) {
+    if (Version.VERSION_1 == version) {
+      return expectedMeasurementsVersion1();
+    } else if (Version.VERSION_2 == version) {
+      return expectedMeasurementsVersion2();
+    } else {
+      LOGGER.error("Not given an expected version. Given version: {}", version);
+      return null;
+    }
+  }
+
+  public static List<Analog> expectedMetaMeasurements() {
+    final List<Analog> measurements = new ArrayList<>();
+    measurements.add(createAnalog("Frequency", 49.98f, UnitSymbol.Hz));
+    measurements.add(createAnalog("Temperature", 12.0f, UnitSymbol.degC));
+    measurements.add(createAnalog("IRMS-N", 0.11f, UnitSymbol.A));
+    return measurements;
+  }
+
+  private static List<Analog> expectedMeasurementsVersion1() {
+    final List<Analog> measurements = new ArrayList<>();
+    measurements.add(createAnalog("U-L1", 0.1f, UnitSymbol.V));
```

Comment:
Waarom gebruik je hier niet de waarden uit de Enums? Nu zijn het 'magic strings'.

Reply:
Even better: why not use org.opensmartgridplatform.adapter.kafka.da.application.mapping.LowVoltageMeasurementDefinition#createAnalog ?

Reply:
I think we want to the test definition as well as the translation of the message. That is why we use "magic" strings in my opinion.